File: Julia/bot.jl
using Discord

const client = Discord.Client("YOUR_BOT_TOKEN_HERE")

function on_ready(client)
    println("Logged in as $(client.user)")
end

function on_message(client, message)
    if message.content == "!hello"
        Discord.send_message(message.channel, "Hello, world!")
    end
end

Discord.register_handler(client, "on_ready", on_ready)
Discord.register_handler(client, "on_message", on_message)

Discord.connect!(client)

File: Nim/bot.nim
import discord, asyncdispatch

proc onReady(client: DiscordClient) {.async.} =
    echo "Logged in as " & client.username

proc onMessage(message: Message) {.async.} =
    if message.content == "!hello":
        await message.reply("Hello World!")

let client = newDiscordClient(token = "YOUR_BOT_TOKEN")
client.onReady = onReady
client.onMessage = onMessage
waitFor client.connect()
